document/FileFormat.mdk

```diff
@@ -2561,23 +2561,28 @@ If this `annotatedCodeLocation` does not occur within a `codeFlow`, the `importa
 
 If present, the `importance` property shall have one of the following values, with the specified meanings:
 
-* `"essential"`: this location is essential for understanding the code flow.
+* `"important"`: this location is important for understanding the code flow.
 
-* `"normal"`: this location is helpful in understanding the code flow.
+* `"essential"`: this location is essential for understanding the code flow.
 
-* `"nonessential"`: this location contributes to a more detailed understanding of the code flow, but is not normally needed. 
+* `"unimportant"`: this location contributes to a more detailed understanding of the code flow, but is not normally needed. 
 
-If this property is absent, it shall be considered to have the value `"normal"`.
+If this property is absent, it shall be considered to have the value `"important"`.
 
 ~ IsoNote
 A viewer might use this property to offer the user three options for viewing a lengthy code flow:
 
-* A "normal view," which omits locations whose `importance` property is `"nonessential"`.
+* A "normal view," which omits locations whose `importance` property is `"unimportant"`.
 * An "abbreviated view," which displays only those locations whose `importance` property is
 `"essential"`.
 * A "verbose view," which displays all the locations in the code flow.
 ~
 
+### `snippet` property { #annotatedCodeLocation-snippet }
+
+An `annotatedCodeLocation` object may contain a property named `snippet` whose value is a string containing
+the text of the source code lines specified by `annotatedCodeLocation.physicalLocation.region`.
+
```

Comment:
Good question. I think I prefer your original proposal. When Jeff originally proposed 'snippet' he explicitly mentioned it including the location of the problem but also some reasonable region around it. This is the way diffing is handled in GitHub.com PRs as well. The 'region' of an error is the code that should be selected in the editor when you double-click, it's the literal pattern that's gone wrong. Consider a spelling error, where 'frz' is not recognized. On double-click you'd want to select that specific thing. The snippet probably would, minimally, include the whole line (so you could see, oh, it is parameter frzIndex of method BlahBlah that is the problem).

For ALCs, the situation is different, there's no specific pattern that comprises the literal problem or which might need to be transmuted by a fix. Instead, things work in the other direction: you choose the snippet that best illustrates the flow entry and describe it in a region. It makes sense that the snippet should match this (i.e., that they exactly relate). Thoughts??